sql-ballerina/tests/complex-query-test.bal

```diff
@@ -273,6 +275,105 @@ function testDateTime() {
     test:assertEquals(value, expected, "Expected record did not match.");
 }
 
+type ResultDates2 record {
+    time:Date DATE_TYPE;
+    time:TimeOfDay TIME_TYPE;
+    time:Civil TIMESTAMP_TYPE;
+    time:Civil DATETIME_TYPE;
+    time:TimeOfDay TIME_TZ_TYPE;
+    time:Civil TIMESTAMP_TZ_TYPE;
+};
+
+@test:Config {
+    groups: ["query", "query-complex-params"]
+}
+function testDateTime2() {
+    MockClient dbClient = checkpanic new (url = complexQueryDb, user = user, password = password);
+    string insertQuery = string `INSERT INTO DateTimeTypes (row_id, date_type, time_type, timestamp_type,
+            datetime_type, time_tz_type, timestamp_tz_type) VALUES (2, '2017-05-23', '14:15:23', '2017-01-25 16:33:55',
+            '2017-01-25 16:33:55', '16:33:55+6:30', '2017-01-25 16:33:55-8:00')`;
+    ExecutionResult? result = checkpanic dbClient->execute(insertQuery);
+    stream<record{}, error?> queryResult = dbClient->query("SELECT date_type, time_type, timestamp_type, datetime_type, "
+            + "time_tz_type, timestamp_tz_type from DateTimeTypes where row_id = 2", ResultDates2);
+    record{| record{} value; |}? data =  checkpanic queryResult.next();
+    record{}? value = data?.value;
+    checkpanic dbClient.close();
+
+    time:Date dateTypeRecord = {year: 2017, month: 5, day: 23};
+    time:TimeOfDay timeTypeRecord = {hour: 14, minute: 15, second:23};
+    time:Civil timestampTypeRecord = {year: 2017, month: 1, day: 25, hour: 16, minute: 33, second: 55};
+    time:TimeOfDay timeWithTimezone = {utcOffset: {hours: 6, minutes: 30}, hour: 16, minute: 33, second: 55, "timeAbbrev": "+06:30"};
```

Comment:
does the actual value have "timeAbbrev" also ?

Reply:
Yes.
https://github.com/ballerina-platform/module-ballerina-sql/blob/5530cc2989f231e6d987e531dfc713a53fdf45b6/sql-native/src/main/java/org/ballerinalang/sql/parameterprocessor/DefaultResultParameterProcessor.java#L532